nemo/collections/nlp/models/language_modeling/megatron_finetune_model.py

```diff
@@ -322,18 +322,23 @@ def inference_step(self, batch, batch_idx, mode, dataloader_idx=0):
             'inputs': input_text,
         }
 
-    def ids_to_text(self, batch_ids):
+    @classmethod
+    def ids_to_text(cls, batch_ids, tokenizer):
         batch_ids = batch_ids.cpu().numpy().tolist()
         texts = []
         for ids in batch_ids:
-            if self.tokenizer.eos_id in ids:
-                idx = ids.index(self.tokenizer.eos_id)
+            if tokenizer.eos_id in ids:
+                idx = ids.index(tokenizer.eos_id)
+                ids = ids[:idx]
+
+            if len(tokenizer.text_to_ids('<extra_id_1>')) == 1 and tokenizer.text_to_ids('<extra_id_1>')[0] in ids:
```

Comment:
Should we be using the T5 sentinel enum class here instead of the hard coded string?

Reply:
Yup just moved the Sentinel class to the T5 model file and imported it in prompt learning and finetuning.